Realm/RLMUtil.mm

```diff
@@ -436,7 +383,68 @@ BOOL RLMIsRunningInPlayground() {
     return [[NSBundle mainBundle].bundleIdentifier hasPrefix:@"com.apple.dt.playground."];
 }
 
-id RLMMixedToObjc(realm::Mixed const& mixed) {
+realm::Mixed RLMObjcToMixed(__unsafe_unretained id v,
+                            __unsafe_unretained RLMRealm *realm,
+                            realm::CreatePolicy createPolicy) {
+    if (!v || v == NSNull.null) {
+        return realm::Mixed();
+    }
+
+    RLMPropertyType type;
+    if ([v conformsToProtocol:@protocol(RLMValue)])
+        type = [v rlm_valueType];
+    else
+        REALM_TERMINATE("Unexpected Type");
```

Comment:
```suggestion
    REALM_ASSERT([v conformsToProtocol:@protocol(RLMValue)]);
    RLMPropertyType type = [v rlm_valueType];
```

if .. else terminate is just a longwinded way of writing an assert assuming it's supposed to be impossible to hit the else case regardless of what the user does. If the user passing an invalid value somewhere can hit this, then it needs to throw an exception instead.